main_integration_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build integration
+
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/magefile/mage/sh"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestSetup tests if Kibana can be run against the current registry
+// and the setup command works as expected.
+func TestSetup(t *testing.T) {
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Chdir(currentDir)
+	err = os.Chdir("testing/environments")
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Make sure services are shut down again at the end of the test
+	defer func() {
+		err = sh.Run("docker-compose", "-f", "snapshot.yml", "-f", "local.yml", "down", "-v")
+		if err != nil {
+			t.Error(err)
+		}
+
+	}()
+	// Spin up services
+	go func() {
+		err = sh.Run("docker-compose", "-f", "snapshot.yml", "pull")
+		if err != nil {
+			t.Error(err)
+		}
+
+		err = sh.Run("docker-compose", "-f", "snapshot.yml", "-f", "local.yml", "up", "--force-recreate", "--remove-orphans", "--build")
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// Check for 5 minutes if service is available
+	for i := 0; i < 5*60; i++ {
+		output, _ := sh.Output("docker-compose", "-f", "snapshot.yml", "-f", "local.yml", "ps")
```

Comment:
Will rerunning the `docker-compose` command cause any issues while previous commands are starting up?

Reply:
If nothing is running, it returns an error. This is the reason I skip the error check. But as this is "only" the `ps` command it should not have any side effects. But I might miss something here.